modules/Students/student_view_details.php

```diff
@@ -2426,9 +2426,15 @@
                             $page->addError(__('Your request failed because you do not have access to this action.'));
                         } else {
                             include './modules/Behaviour/moduleFunctions.php';
+                            
+                            $highestActionBehaviour = getHighestGroupedAction($guid, '/modules/Behaviour/behaviour_view.php', $connection2);
 
                             //Print assessments
-                            echo getBehaviourRecord($container, $gibbonPersonID);
+                            if ($highestActionBehaviour == 'View Behaviour Records_my') {
```

Comment:
In hindsight, can we flip this around, so it's checking the _all permission when it passes no third parameter, otherwise it always passes a third parameter? This is more future-proofed, should there ever be new permissions.